tools/scripts/pytorch-log-parser.py

```diff
@@ -0,0 +1,72 @@
+import json
+import argparse
+import os
+import csv
+
+
+def load_json_files(directory):
+    json_data = {}
+    for root, _, files in os.walk(directory):
+        for file in files:
+            if file.endswith('json'):
+                with open(directory+file, 'r') as f:
+                    data = json.load(f)
+                    json_data.setdefault('traceEvents',[]).append(data['traceEvents'])
+    return json_data
+
+
+def parse(json_file_path, output_file_name):
+
+    data_all = load_json_files(json_file_path)
+    data = data_all['traceEvents']
+    kernels=[]
+
+    for entries in data:
+        for entry in entries:
+            if  'name'in entry and 'cat' in entry and (entry['cat'] == 'kernel' ):
+                kernels.append(entry)
+                print(entry)
+
+    sorted_kernels = sorted(kernels, key=lambda x: ( ['pid']))
+
+    csv_file_name = output_file_name + '.csv'
+    json_file_out = output_file_name + '.json'
+
+    json_data_out = {}
+    json_data_out.setdefault('traceEvents',[]).append({})
+
+    with open(csv_file_name, 'w', newline='') as csvfile:
+        fieldnames = ['pid',  'dur', 'ts']
+        writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+        writer.writeheader()
+
+        i = 1
+        for entry in sorted_kernels:
+            record = {'pid': entry['pid'], 'dur': entry['dur'],
+                     'ts': entry['ts']}
+            json_data_out.setdefault('traceEvents',[]).append(entry)
+
+            writer.writerow(record)
+            if (i) % 8 == 0:
+                csvfile.write('\n')
```

Comment:
Could we orgnaize all the information in database (sqllite), then provide access API ot reorgnize them for renderer ?

Reply:
can you elaborate on what you mean by organizing all information in database format? csv format is more helpful for this report.